.github/workflows/build-test.yml

```diff
@@ -40,7 +40,7 @@ jobs:
 
           echo "COVERPKG: $COVERPKG"
 
-          go test -short -covermode=atomic \
+          GOTOOLCHAIN=go1.25.0+auto go test -short -covermode=atomic \
```

Comment:
Is it required that we use `go1.25.0` or will `go.1.25.3` also suffice the covdata issue? Considering that will be the version we want to test against with the go-toolset using that currently.

side point: would it make more sense if we change the `actions/setup-go`  to using the version in the go.mod file? So we don't need to update version in multiple places everytime
```
      - name: Setup Go
        uses: actions/setup-go@v6
        with:
          go-version-file: 'go.mod'
```